### ISSUE TITLE
[Admin][Permissions] Relationships between enterprises not displayed correctly

### ISSUE THREAD

**Issue by drummer83** [openfoodfoundation/openfoodnetwork#9793]
## Description

This is a follow up of #9768.
While testing #9768 systematically, more issues going beyond #9750 were identified. The examples used the following permissions:
![image](https://user-images.githubusercontent.com/1790176/196057371-71551abb-ea68-4c78-80d8-7436eaefd6d0.png)

The identified issues are:
**!1!** Producer shop should not be displayed as a place to buy producer profile's or another producer shop's or producer hub's products because by definition a producer shop only sells own products.
![image](https://user-images.githubusercontent.com/1790176/196057158-bfd22e57-aa55-4634-9f56-40c2ad6d7e01.png)

**!2!** Producer profile or another producer shop or producer hub should not be displayed as a producer of producer shop's products, because by definition the producer shop only sells own products.
![image](https://user-images.githubusercontent.com/1790176/196057217-a8cb5427-471b-43e0-bac6-d0523cb423f6.png)

**!3!** B granted permission to add to order cycle to A, but A didn't grant permission  to B, so B shouldn't be listed as a place to buy A's products.
![image](https://user-images.githubusercontent.com/1790176/196057250-9679df36-a212-4b78-b98e-be3266818395.png)

These relationships can be seen on the following pages:
1. Shop/hub modal (on page /map or when clicking on a producer in the shopfront)
![image](https://user-images.githubusercontent.com/1790176/196056792-4e09bc0a-7aac-427e-9b70-3f4ba5dac9ce.png)
2. Profile modal (on page /map or when clicking on a producer in the shopfront)
![image](https://user-images.githubusercontent.com/1790176/196056864-f9f5b527-baa3-4688-bb61-8626b5687586.png)
3. Our producers list (on page /shops)
![image](https://user-images.githubusercontent.com/1790176/196056952-0764f3a5-0be7-4cb8-99fe-8d7afc1d15e1.png)
4. Vendors list (on page /producers -> shop for x's products at shop xyz)
![image](https://user-images.githubusercontent.com/1790176/196057048-f34baac5-7c09-4541-b33a-86b37afec3f2.png)
5. Producers list (in the shopfront of an enterprise)
![image](https://user-images.githubusercontent.com/1790176/196057077-41d0772f-533c-4004-a61a-050d336a1cef.png)

A systematical test has been carried out and identified the following problems:
| Permission                                                     | Shop/hub modal | Profile modal | Shops -> our producers | Producers -> Shop for ... | Shop -> Producers |
|-|-|-|-|-|-|
| producer profile grants permission to producer profile          | -              | OK            | OK                     | OK                        | -                 |
| producer profile grants permission to producer shop             | OK             | !1! :x:       | !2! :x:                | !1! :x:                   | !2! :x:           |
| producer profile grants permission to producer hub              | OK             | OK            | OK                     | OK                        | OK                |
| producer profile grants permission to non-producer profile      | -              | OK            | -                      | OK                        | -                 |
| producer profile grants permission to non-producer hub          | OK             | OK            | OK                     | OK                        | OK                |
| producer shop grants permission to producer profile             | OK             | !1! !3! :x:   | OK                     | !1! !3! :x:               | OK                |
| producer shop grants permission to producer shop                | !1! !3! :x:    | -             | !2! :x:                | !1! !3! :x:               | !2! :x:           |
| producer shop grants permission to producer hub                 | !1! !3! :x:    | -             | OK                     | !1! !3! :x:               | OK                |
| producer shop grants permission to non-producer profile         | OK             | OK            | OK                     | OK                        | OK                |
| producer shop grants permission to non-producer hub             | OK             | -             | OK                     | OK                        | OK                |
| producer hub grants permission to producer profile              | OK             | !3! :x:       | OK                     | !3! :x:                   | OK                |
| producer hub grants permission to producer shop                 | !1! !3! :x:    | -             | !2! :x:                | !1! !3! :x:               | !2! :x:           |
| producer hub grants permission to producer hub                  | !3! :x:        | -             | OK                     | !3! :x:                   | OK                |
| producer hub grants permission to non-producer profile          | OK             | OK            | OK                     | OK                        | OK                |
| producer hub grants permission to non-producer hub              | OK             | -             | OK                     | OK                        | OK                |
| non-producer profile grants permission to producer profile      | -              | OK            | -                      | OK                        | -                 |
| non-producer profile grants permission to producer shop         | OK             | OK            | OK                     | OK                        | OK                |
| non-producer profile grants permission to producer hub          | OK             | OK            | OK                     | OK                        | OK                |
| non-producer profile grants permission to non-producer profile  | -              | OK            | -                      | -                         | -                 |
| non-producer profile grants permission to non-producer hub      | OK             | OK            | OK                     | -                         | OK                |
| non-producer hub grants permission to producer profile          | OK             | !3! :x:       | OK                     | !3! :x:                   | OK                |
| non-producer hub grants permission to producer shop             | OK             | !3! :x:       | OK                     | !3! :x:                   | OK                |
| non-producer hub grants permission to producer hub              | !3! :x:        | -             | OK                     | !3! :x:                   | OK                |
| non-producer hub grants permission to non-producer profile      | OK             | OK            | OK                     | OK                        | OK                |
| non-producer hub grants permission to non-producer hub          | OK             | -             | OK                     | OK                        | OK                |

## Expected Behavior

The relationships should be displayed correctly.

## Actual Behaviour

Producer shops are displayed as places to by other producer's products.
Enterprises are displayed as producers of a producer shop's products.
Permissions are applied in both directions while they should only be applied in one direction.

## Steps to Reproduce

1. Set up permissions between the enterprises of interest.
2. Check the 5 places where permissions are displayed in the app.

## Animated Gif/Screenshot

See above.

## Workaround

None.

## Severity
bug-s4: it's annoying, but you can use it

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

**Comment by RachL** [comment]
@drummer83 I'm a bit confused by your screenshots. On the first we can see that A grant permission to B. So with this in mind. Case 1,2,3 are correct?

**Comment by RachL** [comment]
@drummer83 

>  If one of them sold products of another producer it would become a producer hub (sells any) by definition.

By definition, yes. But the display on the profile would still be cases 1,2. The problem here is not the display on the profile. It's just that when a producer sells "own" and gives permission to another shop to sell, we should switch its settings to "any".

The profile display is matching the permissions so these cases look correct to me. 

Ok for case 3. But it's weird I can't reproduce it in production. Perhaps there are more stuff coming into play to reproduce this one?

**Comment by drummer83** [comment]
@RachL I think we both agree that there is something wrong here: either the display or the setting of 'sells'.
Possible solutions to solve this are:
- A: The 'sells' setting overrules the permissions and the display is rendered according to the 'sells' setting.
- B: The permissions overrule/set the 'sells' setting and the display is rendered according to the permissions.

Both is fine for me - no strong opinion.

Re. !3!: Maybe it's the visibility setting of your enterprises?

**Comment by RachL** [comment]
> B: The permissions overrule/set the 'sells' setting and the display is rendered according to the permissions.

@drummer83 but just to be sure, it looks to me that this solution is already in place, right?

**Comment by drummer83** [comment]
Uhmm, not so sure. I didn't focus on this during testing. We definitely don't have the functionality that adding permissions actually sets the 'sells' setting, e.g. from producer shop to producer hub.

**Comment by RachL** [comment]
No but it looks to me like permissions do override the setting and perhaps we should leave it this way, as it's easier for the user. So we could open another issue in order to switch the setting when more permissions are added?

**Comment by drummer83** [comment]
Yes, I will open that as a wishlist item and edit here that !1! and !2! will we covered in that one.
